lib/hydra/derivatives.rb

```diff
@@ -118,7 +118,7 @@ def constantize_processor(processor)
     end
 
     def transform_datastream(file_name, transform_directives, opts={})
```

Comment:
I think empty opts are still valid.  The issue was that the transform_file call from within transform_datastream was not actually passing the opts along.  It was initializing a new empty hash no matter what you passed in.